### PR TITLE
Add new upgrade offer banner alert

### PR DESCRIPTION
**Pull request by stvstnfrd**
This depends on:
- https://github.com/edx/edx-platform/pull/23779

### CODE REVIEW

File: src/offer-alert/OfferAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Alert from '../user-messages/Alert';
+
+function OfferAlert(props) {
```
Comment (stvstnfrd) [inline src/offer-alert/OfferAlert.jsx:6]: This component is nearly identical to `AccessExpiredAlert`.
While we could consolidate (and have like a `RawHtmlAlert`),
we'll soon want to build this HTML directly in the MFE, at which point,
we'll need our own components anyway...

**Review by abutterworth — APPROVED** [review]
LGTM. A thought maybe for a minor future refactor: What do you think about making a `<CourseAlerts courseId={courseId} />` component that encapsulates all the `useXxxxAlert` hooks and the `<AlertList />`?

**Comment by davidjoy** [comment]
I kinda like that idea. 👍

**Comment by stvstnfrd** [comment]
> LGTM. A thought maybe for a minor future refactor: What do you think about making a `<CourseAlerts courseId={courseId} />` component that encapsulates all the `useXxxxAlert` hooks and the `<AlertList />`?

Yeah, I like it!

I also dislike the proliferation of `*-alert` directories popping up in the base `src` directory (of which I'm now 40% responsible!), so those could be grouped together in an `alerts` subdirectory of some sort too.

**Comment by davidjoy** [comment]
Agreed - one thing worth noting is that course home will eventually use some of these alerts, which is why they’re so high up the tree.

**Pull request merged**